circuitbreaker/circuitbreaker.go

```diff
@@ -157,6 +173,9 @@ func (c *circuitbreaker) Run(ctx context.Context, f goresilience.Func) error {
 func (c *circuitbreaker) preDecideState(metricsRec metrics.Recorder) {
 	state := c.getState()
 	switch state {
+	case stateNew:
+		// Close the breaker as this is the first time through and generate a statistic.
+		c.moveState(stateClosed, metricsRec)
```

Comment:
As stated before, the initial state of a circuit breaker is closed, so the above change of `state: stateNew,` and this `moveState` would not be required.

Reply:
True, although it's coupled to my comment above.